packages/ckeditor5-engine/src/dataprocessor/htmldataprocessor.js

```diff
@@ -93,6 +93,21 @@ export default class HtmlDataProcessor {
 		this._domConverter.registerRawContentMatcher( pattern );
 	}
 
+	/**
+	 * If the processor is set to use marked filler, it will insert marked nbsp fillers (`<span data-cke-filler="true">&nbsp;</span>`),
+	 * instead of regular nbsp characters (`&nbsp;`).
+	 *
+	 * This mode allows for more precise handling of block fillers (so they don't leak into editor content) but bloats the editor
+	 * data with additional markup.
+	 *
+	 * This mode may be required by some features and will be turned on by them automatically.
+	 *
+	 * @type {Boolean} value Whether to use (`true`) or not (`false`) marked nbsp fillers.
+	 */
+	useMarkedFiller( value ) {
+		this._domConverter.blockFillerMode = value ? 'markedNbsp' : 'nbsp';
+	}
```

Comment:
`DomConverter` has a third (default) option `'br'` so maybe this should allow passing any `blockFillerMode` to the `DomConverter`?

Reply:
`HtmlDataProcessor` by default uses `nbsp` and does not have any way to change it to `br`, so this is why I don't want to introduce such possibility.